src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -222,238 +209,53 @@
-        // as a string entry for "domain". With the new alias object from the server,
-        // it already has an entry named "domain", which is an integer.
-        // This variable checks for three truths:
-        //   - Does the alias exists?
-        //   - Does it have an entry for "domain"?
-        //   - Is the entry NOT an integer?
-        // If all three of these are true, this user has a legacy label stored locally
-        // that needs to be ported to the "description" entry
-        const storedLegacyAliasLabel =
-          addonRelayAddress &&
-          Object.prototype.hasOwnProperty.call(addonRelayAddress, "domain") &&
-          !Number.isInteger(addonRelayAddress.domain);
-
-        let storedAliasLabel =
-          addonRelayAddress &&
-          Object.prototype.hasOwnProperty.call(addonRelayAddress, "description")
-            ? addonRelayAddress.description
-            : "";
-
-        // Cache the generated_for alias attribute when updating local storage data.
-        // Note that this data attribute only exists in aliases generated through the add-on
-        let storedAliasGeneratedFor = addonRelayAddress?.generated_for ?? "";
-
-        // This covers any legacy label field and remaps them.
-        if (storedLegacyAliasLabel) {
-          storedAliasLabel = addonRelayAddress.domain;
-        }
+      if (localStorageRelayAddresses.length === 0 && existingLocalStorageRelayAddresses) {
+        localStorageRelayAddresses.push(...existingLocalStorageRelayAddresses);
+      }
 
-        // This covers any legacy siteOrigin field and remaps them.
-        const storedLegacyAliasSiteOrigin = addonRelayAddress?.siteOrigin;
-        if (storedLegacyAliasSiteOrigin) {
-          storedAliasGeneratedFor = addonRelayAddress.generated_for;
-        }
+      // Get data from server (Note: All description/generated_for fields should be BLANK)
+      const serverRelayAddresses = await apiRequest(apiRelayAddressesURL);
 
-        const aliasLabelForm = aliasCard.querySelector(
-          "form.relay-email-address-label-form"
-        );
-        const aliasLabelInput = aliasCard.querySelector(
-          "input.relay-email-address-label"
-        );
+      if (localStorageRelayAddresses.length === 0) {
+        // Nothing local! Start fresh! 
+        localStorageRelayAddresses.push(...serverRelayAddresses);
+        browser.storage.local.set({ relayAddresses: localStorageRelayAddresses });
+        return;
+      }
 
-        const aliasLabelWrapper = aliasLabelForm.parentElement;
-        aliasLabelWrapper.classList.add("show-label"); // Field is visible only to users who have the addon installed
+      // WIP/DON'T SHIP: This is a scrape method for the website to grab know label descriptions
+      // Actual fix should be to interacte with webpage local storage (however, localStorage does not have an labels stored currently) 
+      const masks = document.querySelectorAll("div[class*='Alias_controls']");
```

Comment:
Question for @Vinnl: What is the best way to store the label field for masks when relying on local storage?

This approach still scrapes the page (but only for labels), when on dashboard page.

Reply:
To answer that, I submitted a PR with docs on it: https://github.com/mozilla/fx-private-relay/pull/1953

As documented, the add-on should already be storing the label field when relying on local storage, so I'm guessing that something in the approach described there has a bug?